src/react/types.ts

```diff
@@ -119,24 +119,22 @@ export type UseMutateAsyncFunction<
   options?: MutateOptions<TData, TError, TVariables, TContext>
 ) => Promise<TData>
 
-export interface UseMutationResult<
+export type UseBaseMutationResult<
   TData = unknown,
   TError = unknown,
   TVariables = unknown,
   TContext = unknown
-> {
-  context: TContext | undefined
-  data: TData | undefined
-  error: TError | null
-  failureCount: number
-  isError: boolean
-  isIdle: boolean
-  isLoading: boolean
-  isPaused: boolean
-  isSuccess: boolean
+> = Exclude<
+  MutationObserverResult<TData, TError, TVariables, TContext>,
+  'mutate'
+> & {
```

Comment:
@TkDodo I'm trying to override the type of mutate here and define it with UseMutateFunction so I can fix this type error

<img width="1172" alt="Screenshot 2021-09-08 at 11 09 12 PM" src="https://user-images.githubusercontent.com/28605803/132535701-8916ce37-2ad0-4969-a9df-58bfe2e61e78.png">

I tried to used Omit, but it broke the union in MutationObserverResult and I can't change it to interface and then extend as MutationObserverResult is a type 

do you have any idea what the best approach would be here?